presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -307,21 +308,26 @@ private PagesIndexOrdering createPagesIndexComparator(List<Integer> sortChannels
 
     public Supplier<LookupSource> createLookupSourceSupplier(Session session, List<Integer> joinChannels)
     {
-        return createLookupSourceSupplier(session, joinChannels, Optional.empty(), Optional.empty());
+        return createLookupSourceSupplier(session, joinChannels, Optional.empty(), Optional.empty(), Optional.empty());
     }
 
     public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel)
     {
+        return createPagesHashStrategy(joinChannels, hashChannel, Optional.empty());
+    }
+
+    public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel, Optional<List<Integer>> outputChannels)
```

Comment:
I prefer to leave it optional here for now than modifying all classes that use `PagesIndex` and `JoinCompiler`. It will be easier to rebase and resolve conflicts while exposing the parameter.